data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/impl/consumer/OrderedConsumerVerticle.java

```diff
@@ -186,16 +199,24 @@ private void recordsHandler(KafkaConsumerRecords<Object, CloudEvent> records) {
     // Put records in internal per-partition queues.
     for (int i = 0; i < records.size(); i++) {
       final var record = records.recordAt(i);
-      final var executor = executorFor(new TopicPartition(record.topic(), record.partition()));
-      executor.offer(() -> dispatch(record));
+      final var recordContext = new ConsumerRecordContext(record); // resets timer to start calculating the executor queue latency
+
+      TopicPartition topicPartition = new TopicPartition(record.topic(), record.partition());
+      final var executor = executorFor(topicPartition);
+      executor.offer(() -> dispatch(record, recordContext));
+
+      setValueQueueLength(topicPartition, executor.getQueue());
```

Comment:
Can we record these new metrics only when a feature flag is enabled similar to the rate limiter `egress.getFeatureFlags().getEnableRateLimiter()`?

I believe these 2 metrics might have a performance impact, especially on topics with many partitions